auslib/test/blobs/test_gmp.py

```diff
@@ -59,6 +59,65 @@ def setUp(self):
 }
 """)
 
+    def testValidateHashValue(self):
+        blob = GMPBlobV1()
+        blob.whitelistedDomains = {"a.com": ('gg',), 'boring.com': ('gg',), 'nice.com': ('gg',)}
+        blob.loadJSON("""
+{
+    "name": "validName",
+    "schema_version": 1000,
+    "hashFunction": "SHA512",
+    "vendors": {
+        "a": {
+            "version": "1",
+            "platforms": {
+                "p": {
+                    "filesize": 2,
+                    "hashValue": "3",
+                    "fileUrl": "http://a.com/blah"
+                },
+                "q": {
+                    "filesize": 4,
+                    "hashValue": "5",
+                    "fileUrl": "http://boring.com/blah"
+                },
+                "r": {
+                    "alias": "q"
+                }
+            }
+        },
+        "b": {
+            "version": "5",
+            "platforms": {
+                "q": {
+                    "filesize": 10,
+                    "hashValue": "11",
+                    "fileUrl": "http://boring.com/foo"
+                },
+                "r": {
+                    "filesize": 666,
+                    "hashValue": "666",
+                    "fileUrl": "http://nice.com/fire"
+                },
+                "default": {
+                    "filesize": 20,
+                    "hashValue": "50",
+                    "fileUrl": "http://boring.com/bar"
+                }
+            }
+        }
+    }
+}
+""")
+        self.assertRaises(Exception, blob.validate, 'gg', self.whitelistedDomains)
+        # Filling the 'hashValue' fields with correct length hashValues
+        for vendor in blob["vendors"]:
+            if "platform" in blob["vendors"][vendor]:
+                for platform in blob["vendors"][vendor]["platforms"]:
+                    if "hashValue" in blob["vendors"][vendor]["platforms"][platform]:
+                        blob["vendors"][vendor]["platforms"][platform]["hashValue"] = 'K' * 128
```

Comment:
It's usually a bad idea to put code like this in a test if it's not necessary (you run the risk of causing test failures for something unrelated to the test). Instead, run this code separately and copy/paste the result into each hashValue above.

Probably best to use an actual hex character too :). "K" is not valid in a SHA hash.